src/transformStateWithClones.js

```diff
@@ -8,6 +8,37 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const clonedState = [];
+  const currentStateData = { ...state };
+
+  function addDataToArray() {
+    clonedState.push({ ...currentStateData });
+  };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(currentStateData, { ...action.extraData });
```

Comment:
no need to use copy here

```suggestion
        Object.assign(currentStateData, action.extraData );
```